libs/exec/exec.go

```diff
@@ -0,0 +1,212 @@
+package exec
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	osexec "os/exec"
+)
+
+type Executor struct {
+	interpreter interpreter
+	dir         string
+	scriptFiles []string
+}
+
+func NewCommandExecutor(dir string) (*Executor, error) {
+	interpreter, err := findInterpreter()
+	if err != nil {
+		return nil, err
+	}
+	return &Executor{
+		interpreter: interpreter,
+		dir:         dir,
+		scriptFiles: nil,
+	}, nil
+}
+
+func (e *Executor) StartCommand(ctx context.Context, command string) (func() error, io.Reader, error) {
+	e.interpreter.prepare(command)
+	return e.start(ctx)
+}
+
+func (e *Executor) start(ctx context.Context) (func() error, io.Reader, error) {
+	e.scriptFiles = append(e.scriptFiles, e.interpreter.getScriptFile())
+	cmd := osexec.CommandContext(ctx, e.interpreter.getExecutable(), e.interpreter.getArgs()...)
+	cmd.Dir = e.dir
+
+	outPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	errPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cmd.Wait, io.MultiReader(outPipe, errPipe), cmd.Start()
```

Comment:
In the previous comment I meant returning them separately and letting the caller use `io.MultiReader` is they want to. Can be done in a follow up PR.